ingest/sqs/domain/service.go

```diff
@@ -0,0 +1,52 @@
+package domain
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	poolmanagertypes "github.com/osmosis-labs/osmosis/v24/x/poolmanager/types"
+)
+
+// BlockPoolUpdateTracker is an interface for tracking
+// the pools that were updated in a block.
+// It persists the pools using "Track" methods in its internal state.
+// It tracks the latest pool update, discarding the previous updates.
+// Only on Reset, the internal state is cleared.
+type BlockPoolUpdateTracker interface {
+	// TrackConcentrated tracks the concentrated pool.
+	TrackConcentrated(pool poolmanagertypes.PoolI)
+
+	// TrackConcentratedPoolIDTickChange tracks the concentrated pool ID tick change.
+	// Due to internal implementation, it is non-trivial to apply tick changes.
+	// As a result, we track the pool ID tick change and read the pool with all of its ticks
+	// if at least one tick change was applied within the block.
+	TrackConcentratedPoolIDTickChange(poolID uint64)
+
+	// TrackCFMM tracks the CFMM pool.
+	TrackCFMM(pool poolmanagertypes.PoolI)
+
+	// TrackCosmWasm tracks the CosmWasm pool.
+	TrackCosmWasm(pool poolmanagertypes.PoolI)
+
+	// GetConcentratedPools returns the tracked concentrated pools.
+	GetConcentratedPools() []poolmanagertypes.PoolI
+
+	// GetConcentratedPoolIDTickChange returns the tracked concentrated pool ID tick change.
+	GetConcentratedPoolIDTickChange() map[uint64]struct{}
+
+	// GetCFMMPools returns the tracked CFMM pools.
+	GetCFMMPools() []poolmanagertypes.PoolI
+
+	// GetCosmWasmPools returns the tracked CosmWasm pools.
+	GetCosmWasmPools() []poolmanagertypes.PoolI
+
+	// Reset clears the internal state.
+	Reset()
+}
+
+// NodeStatusChecker is an interface for checking the node status.
+type NodeStatusChecker interface {
+	// IsNodeSyncing checks if the node is syncing.
+	// Returns true if the node is syncing, false otherwise.
+	// Returns error if the node syncing status cannot be determined.
+	IsNodeSyncing(ctx sdk.Context) (bool, error)
+}
```

Comment:
Ah I see, maybe we can add this clarification in the godoc of the IsNodeSyncing method, I understand now after reading it here.

Reply:
The godoc says
https://github.com/osmosis-labs/osmosis/blob/ddc566b82e90cd0f17774c2aab51a98854c56b3a/ingest/sqs/service/node_status_checker.go#L23

Same as the mock:
https://github.com/osmosis-labs/osmosis/blob/ddc566b82e90cd0f17774c2aab51a98854c56b3a/ingest/sqs/domain/mocks/node_status_checker_mock.go#L21

Instead of duplicating the spec, I prefer to reference the interface that documents the details so that we only need to update the spec in one place and prevent it from going out-of-sync across different implementations.

Given justification, please let me know if there is anything actionable you recommend here